eth/backend.go

```diff
@@ -604,46 +617,38 @@ func (s *Ethereum) handleRemovals(parlia *parlia.Parlia, nonce uint64) error {
 	// Filter out node IDs that are in the add set
 	nodeIDsToRemove := make([]enode.ID, 0, len(s.config.EVNNodeIDsToRemove))
 	for _, id := range s.config.EVNNodeIDsToRemove {
-		if _, exists := addSet[id]; !exists {
-			nodeIDsToRemove = append(nodeIDsToRemove, id)
+		if _, exists := registeredSet[id]; exists {
+			if _, exists := addSet[id]; !exists {
+				nodeIDsToRemove = append(nodeIDsToRemove, id)
+			} else {
+				log.Debug("Skipping node ID removal", "id", id, "reason", "also in EVNNodeIDsToAdd")
+			}
 		} else {
-			log.Debug("Skipping node ID removal", "id", id, "reason", "also in EVNNodeIDsToAdd")
+			log.Debug("Skipping node ID removal", "id", id, "reason", "not registered")
 		}
 	}
 
 	if len(nodeIDsToRemove) == 0 {
+		log.Debug("No node IDs to remove after filtering")
 		return nil
 	}
 
 	trx, err := parlia.RemoveNodeIDs(nodeIDsToRemove, nonce)
 	if err != nil {
 		return fmt.Errorf("failed to create node ID removal transaction: %v", err)
 	}
-	if err := s.txPool.Add([]*types.Transaction{trx}, false); err != nil {
-		return fmt.Errorf("failed to add node ID removal transaction to pool: %v", err)
+	if errs := s.txPool.Add([]*types.Transaction{trx}, false); len(errs) > 0 {
```

Comment:
if the `errs = [nil]`, it will return a error too. How about this:

```go
if errs := s.txPool.Add([]*types.Transaction{trx}, false); len(errs) >0 && errs[0] != nil {
```